### PR TITLE
fix(auth): disable automatic universe domain check for MDS

### PR DESCRIPTION
**Pull request by quartzmo**

### CODE REVIEW

File: auth/grpctransport/grpctransport.go [hunk 1]
```diff
@@ -337,17 +337,19 @@ func (c *grpcCredentialsProvider) getClientUniverseDomain() string {
...
-		return nil, err
-	}
-	if err := transport.ValidateUniverseDomain(c.getClientUniverseDomain(), credentialsUniverseDomain); err != nil {
-		return nil, err
-	}
 	token, err := c.creds.Token(ctx)
 	if err != nil {
 		return nil, err
 	}
+	if source, _ := token.Metadata["auth.google.tokenSource"].(string); source != "compute-metadata" {
```
Comment (codyoss) [inline auth/grpctransport/grpctransport.go:344]: These maps could be nil. Can you add a check.

Comment (quartzmo) [reply auth/grpctransport/grpctransport.go:344]: 1. Would it be better to always initialize with an empty map?
2. Should I add the check to this existing usage as well? https://github.com/googleapis/google-cloud-go/blob/main/auth/grpctransport/directpath.go#L69-L74

Comment (codyoss) [reply auth/grpctransport/grpctransport.go:344]: Users can create their own token types that we use, so best to be defensive.

Comment (quartzmo) [reply auth/grpctransport/grpctransport.go]: Updated, PTAL.

File: auth/auth.go [hunk 2]
```diff
@@ -101,6 +101,20 @@ func (t *Token) IsValid() bool {
 	return t.isValidWithEarlyExpiry(defaultExpiryDelta)
 }
 
+// MetadataString is a convenience method for accessing string values in the
+// token's metadata. Returns an empty string if the metadata is nil or the value
+// for the given key cannot be cast to a string.
+func (t *Token) MetadataString(k string) string {
```
Comment (codyoss) [inline auth/auth.go:107]: Great idea!

**Review by codyoss — APPROVED** [review]

**Pull request merged**